azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+			},
+		},
+	}
+
+	return &results
+}
+
+func flattenVideoAnalyzerStorageAccounts(input *[]videoanalyzer.StorageAccount) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, storageAccount := range *input {
+		output := make(map[string]interface{})
+
+		if storageAccount.ID != nil {
+			output["id"] = *storageAccount.ID
+		}
+
+		if storageAccount.Identity != nil {
+			output["identity_id"] = *storageAccount.Identity.UserAssignedIdentity
+		}
+
+		results = append(results, output)
+	}
+
+	return results
+}
+
+func expandAzureRmVideoAnalyzerIdentity(d *pluginsdk.ResourceData) (*videoanalyzer.Identity, error) {
+	identityRaw := d.Get("identity").([]interface{})
+	if identityRaw[0] == nil {
+		return nil, nil
+	}
+	identity := identityRaw[0].(map[string]interface{})
+	result := &videoanalyzer.Identity{
+		Type: utils.String(identity["type"].(string)),
+	}
+	var identityIdSet []interface{}
+	if identityIds, exists := identity["identity_ids"]; exists {
+		identityIdSet = identityIds.(*pluginsdk.Set).List()
+	}
+
+	// If type contains `UserAssigned`, `identity_ids` must be specified and have at least 1 element
+	if *result.Type == "UserAssigned" {
+		if len(identityIdSet) == 0 {
+			return nil, fmt.Errorf("`identity_ids` must have at least 1 element when `type` includes `UserAssigned`")
+		}
+
+		userAssignedIdentities := make(map[string]*videoanalyzer.UserAssignedManagedIdentity)
+		for _, id := range identityIdSet {
+			userAssignedIdentities[id.(string)] = &videoanalyzer.UserAssignedManagedIdentity{}
+		}
+		result.UserAssignedIdentities = userAssignedIdentities
+	} else if len(identityIdSet) > 0 {
+		// If type does _not_ contain `UserAssigned` (i.e. is set to `SystemAssigned` or defaulted to `None`), `identity_ids` is not allowed
+		return nil, fmt.Errorf("`identity_ids` can only be specified when `type` includes `UserAssigned`; but `type` is currently %q", *result.Type)
+	}
+
+	return result, nil
+}
+
+func flattenAzureRmVideoServiceIdentity(identity *videoanalyzer.Identity) ([]interface{}, error) {
+	if identity == nil {
+		return make([]interface{}, 0), nil
+	}
+
+	result := make(map[string]interface{})
+	result["type"] = identity.Type
+
+	identityIds := make([]interface{}, 0)
+	if identity.UserAssignedIdentities != nil {
+		/*
+		   "userAssignedIdentities": {
+		     "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1": {
+		       "principalId": "00000000-0000-0000-0000-000000000000",
+		       "clientId": "00000000-0000-0000-0000-000000000000"
+		     },
+		   }
+		*/
+		for key := range identity.UserAssignedIdentities {
+			parsedId, err := msiparse.UserAssignedIdentityID(key)
+			if err != nil {
+				return nil, err
+			}
+			identityIds = append(identityIds, parsedId.ID())
+		}
+		result["identity_ids"] = identityIds
+	}
+
+	return []interface{}{result}, nil
```

Comment:
Since this is a pointer in the SDK but needs to be a string, we'll need to conditionally pull this out incase it's nil (for `type`) - we also need to ensure the field `identity_ids` always has a value since this is a nested object:

```suggestion
	identityType := ""
	if identity.Type != nil {
		identityType = *identity.Type
	}

	identityIds := make([]interface{}, 0)
	if identity.UserAssignedIdentities != nil {
		/*
		   "userAssignedIdentities": {
		     "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1": {
		       "principalId": "00000000-0000-0000-0000-000000000000",
		       "clientId": "00000000-0000-0000-0000-000000000000"
		     },
		   }
		*/
		for key := range identity.UserAssignedIdentities {
			parsedId, err := msiparse.UserAssignedIdentityID(key)
			if err != nil {
				return nil, err
			}
			identityIds = append(identityIds, parsedId.ID())
		}
	}

	return []interface{}{
		map[string]interface{}{
			"type": identityType,
			"identity_ids": identityIds,
		},
	}, nil
```